File: day1_part1_2022.java

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class solution {
    public static void main(String[] args) {
        int maxCalories = 0;
        int currentCalories = 0;

        try (BufferedReader br = new BufferedReader(new FileReader("input.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                if (line.isEmpty()) {
                    if (currentCalories > maxCalories) {
                        maxCalories = currentCalories;
                    }
                    currentCalories = 0;
                    continue;
                }

                int calories = Integer.parseInt(line);
                currentCalories += calories;
            }

            if (currentCalories > maxCalories) {
                maxCalories = currentCalories;
            }

            System.out.println(maxCalories);

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: day1_part1_2022.fs
module Day1

let input = System.IO.File.ReadAllLines("input.txt")
let mutable currentCalories = 0
let mutable maxCalories = 0

for line in input do
    if line = "" then
        if currentCalories > maxCalories then
            maxCalories <- currentCalories
        currentCalories <- 0
    else
        currentCalories <- currentCalories + int line

if currentCalories > maxCalories then
    maxCalories <- currentCalories

printfn "%d" maxCalories